oioswift/common/middleware/container_hierarchy.py

```diff
@@ -43,6 +43,55 @@
 
 
 class RedisDb(object):
+    lua_script_hkeys = """
+        local get_index = function(key, start, delimiter)
+            -- Get index of the first delimiter of the key
+            local index = string.find(key, delimiter, start)
+            if index  == nil then
+                return #key
+            end
+            return index
+        end
+        local get_first_level = function(key, start, delimiter)
+            -- Get the hierarchy first level
+            -- start = start looking at this index
+            -- delimiter = used for separating level
+            -- key = prefix .. delimiter .. first_level ...
+            -- Examples get_first_level("aaa/bbb/ccc", 4, '/') = "aaa/bbb"
+            -- get_first_level("aaa/bbb/ccc", 8, '/') = "aaa/bbb/ccc"
+
+            local index = get_index(key, start, delimiter)
+            local rv = string.sub(key, 0, index)
```

Comment:
The character at index `index` is included in the substring. The example should be `get_first_level("aaa/bbb/ccc", 4, '/') = "aaa/bbb/"`.